src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -73,17 +75,25 @@ def config_dashboard(self, daemon_descrs: List[DaemonDescription]):
             service_url
         )
 
+
 class AlertmanagerService(CephadmService):
     TYPE = 'alertmanager'
     DEFAULT_SERVICE_PORT = 9093
 
     def create(self, daemon_spec: CephadmDaemonSpec) -> str:
         assert self.TYPE == daemon_spec.daemon_type
+        assert daemon_spec.spec
         return self.mgr._create_daemon(daemon_spec)
 
-    def generate_config(self, daemon_spec: CephadmDaemonSpec) -> Tuple[Dict[str, Any], List[str]]:
+    def generate_config(self, daemon_spec: CephadmDaemonSpec[AlertManagerSpec]) -> Tuple[Dict[str, Any], List[str]]:
         assert self.TYPE == daemon_spec.daemon_type
-        deps = [] # type: List[str]
+        deps = []  # type: List[str]
```

Comment:
```suggestion
        deps: List[str] = []
```

cephadm is not getting backported to py2 :tada: